Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -435,14 +505,14 @@ public final class RegistryClient {
                                 })
                             }
                         } catch {
-                            completion(.failure(error))
+                            completion(.failure(RegistryError.failedToDetermineExpectedChecksum(error)))
```

Comment:
I think "failedToComputeChecksum" is more accurate. We do have the expected checksum (i.e., `expectedChecksum` from local storage or registry API) and we are trying to compute checksum of the archive to see if it matches.